### DISCUSSION TITLE
[UI/UX] Pin the bottommost object in the tree view if it is located outside the screen area. (feature request)

### DISCUSSION THREAD

**Discussion by kaiwas** [FreeCAD/FreeCAD#16999] [Discussions that will turn in to issues but need refining]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

At the moment, if the tree view is very tall and does not fit into the screen area, you have to constantly scroll to newly created objects (for example, to then move them to a group located at the top).

I have a proposal to allocate an area at the bottom of the panel for one object that will always be visible, even if the tree view is very tall.

When the tree view is large, a separator is created that separates the bottom object from the rest of the tree.

If we scroll the tree view, the separator at the bottom disappears when the object enters the real visibility zone. But at the same time, when the top of the tree view goes beyond the screen, a separator-identifier appears at the top, which simply shows that the tree view is greater than the screen height.

UPD/ If the scroll slider is in the middle of a tall tree, then two dividers are displayed (at the bottom and at the top).
This way, we always have access to new objects and the desired area of ​​the tree view to which we have scrolled.

The concept is in the picture.
![TreeeVIEW](https://github.com/user-attachments/assets/6af96079-f237-460c-b644-6f49662da3d0)

### Full version info

```shell
OS: Manjaro Linux (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 1.1.0dev.38871 (Git) AppImage
Build type: Release
Branch: main
Hash: 813ee36cb80e0d29c2c9b991a7b2fee2c61f5cf6
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: Russian/Russia (ru_RU)
Stylesheet/Theme/QtStyle: OpenLight.qss/OpenLight/Qt default
Installed mods: 
  * lattice2 1.0.0
  * PitchedRoof
  * OpenTheme 2024.9.1
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by furgo16** [comment]
It took me a while to understand the proposal from the title. I would suggest a small change in the issue title:

```diff
- Fix the bottommost object
+ Pin the bottommost object
```

Otherwise it reads "to fix" as in "to repair" something, whereas the intended meaning is to keep the bottom object always visible at the bottom of the panel view.
